README.md

```diff
@@ -1,29 +1,78 @@
 
-http verbs mean something urls mean something
+4. Next the server sends a response back to the client, this does not necessarily mean that the newsletter was posted, the response could be that there was an error posting or something like that. However in this case we will say that the newsletter post went through properly, so the server sends a success message and tells the browser what page to go to and render. 
 
-put it all together and we get predictable guessable urls that describe the things they are representing. we call these things a resource.
+5. Lastly the browser receives the server information and gives the user feedback, in this case it shows you a message saying that your newsletter was successfully posted.
 
-/posts vs /pictures
 
-the 7 restful routes 7 restful actions
+## RESTful Conventions in Rails
 
-REST is a concept - not a technology. for right now, just think of it as standard URLS and HTTP verbs that describe what they are trying to do
+Rails has RESTful principles built into its core, so whether you are utilizing the built in view rending system or using the application purely as API you will have the tools necessary to follow standardized routing procedures.
 
-diagram that shows the mirroring of CRUD, to ROutes to controller Actions
+If you were to build out a feature in Rails that entails all of the potential RESTful routes, you would have seven different RESTful URLs, HTTP verbs (more on those later), and actions. As an example, imagine that you have a Newsletter module in your application, if you run ```rake routes``` in the terminal you will receive the following output:
+
+```
+newsletters     GET     /newsletters(.:format)          newsletters#index
+                POST    /newsletters(.:format)          newsletters#create
+new_newsletter  GET     /newsletters/new(.:format)      newsletters#new
+edit_newsletter GET     /newsletters/:id/edit(.:format) newsletters#edit
+newsletter      GET     /newsletters/:id(.:format)      newsletters#show
+                PATCH   /newsletters/:id(.:format)      newsletters#update
+                PUT     /newsletters/:id(.:format)      newsletters#update
+                DELETE  /newsletters/:id(.:format)      newsletters#destroy
+```
+
+Rails does a great job of integrating RESTful routes into their system, if you can understand routes in Rails you can understand REST in general. Going through the output above you should be able to see that all of the potential CRUD actions are all there, from querying all of the records to deleting a single item from the database, and all of the actions are wired up using RESTful routing nomenclature.
+
+Here is a diagram that shows how how the views, controller actions, routes, and HTTP verbs are all mapped together
+
+![REST Diagram](http://reif.io/lib/flatiron/rest_diagram.png)
```

Comment:
This diagram is good except I would combine the method and the path into just one box each.